feat: add flake support

ayys opened the pull request:
This PR adds flake support to star which allows users of the Nix package manager or NixOS to run `nix build` to build the package, and `nix develop` to drop into a shell with all the dependencies installed.

ayys committed:
feat: add flake support

this commit adds flake support to star which allows users of the Nix
package manager or NixOS to run `nix build` to build the package, and
`nix develop` to drop into a shell with all the dependencies
installed.

ayys commented:
e.g. here is how I was able to add star to my config: https://github.com/ayys/dots/commit/46360fa14438d929e4de2f35fd398615f116171c

I'd love to contribute to the README if you would like me to, a section on how to install star with nix.

Fruchix commented:
Hi, thank you for the PR!
I've never used Nix, so I'll take a bit of time before accepting the PR: I want to be able to understand how it works and to make sure that everything is fine.

But basically its seems like a really good idea that I'm keen on adopting. Updating the README would be great, maybe a foldable `Installation on nix` section under the installation from release?

If you prefer to wait for me to review the files before updating the README then do so, I'll keep you updated.

One question that I can raise wihtout much Nix knowledge, is that you committed the `.envrc` file. Is it really needed? Same form `flake.lock`?

Do you know of some tools that support Nix in the same way that you suggest? So I can check how it was done in those projects.

ayys committed:
chore: remove .envrc from the repo

this is not really needed

ayys commented:
> you committed the .envrc file. Is it really needed?

It is not strictly needed and can be removed.
A lot of developers have adopted [direnv](https://direnv.net/) as a way to quickly provide development environments. With the `.envrc` file, the flake-based development environment will automatically be "initialized" upon CDing into the directory.

> Same form flake.lock?

This is definitely necessary. Here is the rationale from [zero-to-nix docs](https://zero-to-nix.com/concepts/flakes/#lockfile):
> The flake.lock file ensures that Nix flakes have purely deterministic outputs. A flake.nix file without an accompanying flake.lock should be considered incomplete and a kind of proto-flake. Any Nix CLI command that is run against the flake—like nix build, nix develop, or even nix flake show—generates a flake.lock for you.

ayys committed:
feat: add a section "Installation on Nix" in readme

Fruchix commented:
Currently you are building from the sources, however for a more reproducible installation, wouldnt it be better to build on a specific tag?
Say for example v2.0.0 currently, then update the flake files when we add a new tag.

Fruchix commented on README.md:
```diff
@@ -386,6 +386,27 @@ echo '[[ ":${PATH}:" =~ ":/usr/local/bin:" ]] || export PATH="/usr/local/bin:$PA
+nix run github:Fruchix/star
+
+# Install to your user profile
+nix profile install github:Fruchix/star
```
Apparently, `nix profile install` is deprecated alias for `nix profile add`.

Is it better to keep using `install` for backward compatibility?

ayys replied:
`add` and `install` are [aliases](https://nix.dev/manual/nix/2.32/command-ref/new-cli/nix3-profile-add.html), and yes, `install` is better for backwards compatibility.

ayys committed:
feat: peg to nix 25.11

Actually, it makes sense to peg it to nix 25.11 instead of unstable
because unstable tends to change quite a bit.

ayys commented:
> however for a more reproducible installation, wouldnt it be better to build on a specific tag?

oh definitely! the way you build from tags is like this:
```
nix run github:Fruchix/star/v2.0.0 # <-- notice the git tag at the end
```

ofcourse, that tag needs to have flake.nix and flake.lock defined. 
Here is the output when I run with tag `v2.0.0`:

```
[ayys@ayys:~]$ nix run github:Fruchix/star/v2.0.0
error: path '«github:Fruchix/star/dcf548cd5c8ce092ba11028a3b26259e99856a78»/flake.nix' does not exist
```
